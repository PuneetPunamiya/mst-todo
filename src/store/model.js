import { types, unprotect } from "mobx-state-tree"

const Todo = types.model({
  task: types.string,
  done: types.optional(types.boolean, false)
})
.actions(self => ({

  toggle() {
      self.done = !self.done
  }
}))

export const TodoStore = types.model({
  todos: types.array(Todo)
})

const s = TodoStore.create({ todos: [] })
unprotect(s) // needed to allow modifying outside of an action
s.todos.push({ task: "Grab coffee" })
// console.log(s.todos[0]) // prints: "Grab coffee"