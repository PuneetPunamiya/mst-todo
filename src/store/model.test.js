import {TodoStore} from './model'
import {unprotect} from 'mobx-state-tree'


it("shoult toggle", () => {
  const s = TodoStore.create({ todos: [] })
  unprotect(s)
  s.todos.push({ task: "asd" })
  // expect(s.todos[0].done).toBe(false)
  s.todos[0].toggle()
  expect(s.todos[0].done).toBe(true)
})

it("should test with snapshots", () => {
  const s = TodoStore.create({ todos: [] })
  unprotect(s)
  s.todos.push({ task: "asd" })
  // expect(s.todos[0].done).toBe(false)
  s.todos[0].toggle()
  expect(s).toMatchSnapshot();
})