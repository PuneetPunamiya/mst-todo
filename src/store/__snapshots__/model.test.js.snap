// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test with snapshots 1`] = `
Object {
  "todos": Array [
    Object {
      "done": true,
      "task": "asd",
    },
  ],
}
`;
